docs/2.9.1/docs/app-dev/explicit-contract-disclosure.rst

```diff
@@ -264,3 +264,23 @@ These last two steps are executed using the new Daml Script functions supporting
 
 .. note:: For an example using Java bindings for client applications, see the
   `Java Bindings StockExchange example project <https://github.com/digital-asset/ex-java-bindings/blob/f474ae83976b0ad197e2fabfce9842fb9b3de907/StockExchange/README.rst>`_.
+
+Safeguards
+------------------------------------------
+Explicit Contract Disclosure usage should always be accompanied by on-ledger contracts. This ensures that workflows executed based on the disclosed contract's contents conform to on-ledger agreements between the stakeholders (or trusted parties) involved.
```

Comment:
@juliuswilliam-da We might need to adapt the wording here. Usage of `contracts` noun here is slightly confusing. Are you referring to the actual Daml contracts that are instances of Daml templates that are stored on ledger? Or to preconditions for executing choices?

Reply:
Good point, I am referring to the `contracts` noun here.
What would be the correct terminology for `contracts` in public-facing content?